cdap-client-tests/src/test/java/co/cask/cdap/client/ApplicationClientTestRun.java

```diff
@@ -120,6 +123,21 @@ public void testAll() throws Exception {
   }
 
   @Test
+  public void testAppConfig() throws Exception {
+    ConfigTestApp.ConfigClass config = new ConfigTestApp.ConfigClass("testStream", "testDataset");
+    appClient.deploy(Id.Namespace.DEFAULT, createAppJarFile(ConfigTestApp.class), GSON.toJson(config));
```

Comment:
it would be better to also allow somebody to just pass in the config object, then we turn it into json. otherwise people need gson or something like it in their tests.